pkg/http-handler/http-handler.go

```diff
@@ -74,6 +75,25 @@ func CreateAppRepository(handler kube.AuthHandler) func(w http.ResponseWriter, r
 	}
 }
 
+// ValidateAppRepository returns a 200 if the connection to the AppRepo can be established
+func ValidateAppRepository(handler kube.AuthHandler) func(w http.ResponseWriter, req *http.Request) {
```

Comment:
I think you should be able to add unit tests for this handler by creating an `server := httptest.NewServer(...)` that just returns the index.yaml. Then your test just uses `server.URL` of the test server in the request body for the repo URL?

The test here only needs to verify the 200 response (and response body) when the test server also responds with a 200, and the error response when it does not.